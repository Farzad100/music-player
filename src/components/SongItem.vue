<!-- eslint-disable linebreak-style -->
<!-- eslint-disable vuejs-accessibility/click-events-have-key-events -->
<template>
   <li class="flex justify-between items-center p-3 pl-6 cursor-pointer transition
            duration-300 hover:bg-gray-50 snapshot" :class="`song-id-${song.id}`">
            <div>
              <router-link :to="{ name: 'song', params: { id: song.id }}"
              class="font-bold block text-gray-600 composition-name">
                {{song.name}}
              </router-link>
              <span class="text-gray-500 text-sm">{{song.genre}}</span>
            </div>

            <div class="text-gray-600 text-lg">
              <router-link custom
              :to="{ name: 'song', params: { id: song.id }, hash: '#comments'}"
              v-slot="{ navigate }">
                <span class="comments" @click="navigate">
                  <i class="fa fa-comments text-gray-600"></i>
                  {{ comment_count }}
                </span>
              </router-link>
            </div>
          </li>
</template>
<!-- eslint-disable linebreak-style -->
<script>
import { mapState } from 'vuex';

export default {
  name: 'SongItem',
  data() {
    return {
      comment_count: 0,
    };
  },
  props: {
    index: {
      type: Number,
      required: true,
    },
    song: {
      type: Object,
      required: true,
    },
  },
  computed: {
    ...mapState(['comments']),
  },
  created() {
    this.comments.forEach((comment) => {
      if (this.song.id === comment.sid) {
        this.comment_count += 1;
      }
    });
  },
};
</script>
