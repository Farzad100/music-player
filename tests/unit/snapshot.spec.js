/* eslint-disable linebreak-style */
import SongItem from '@/components/SongItem.vue';
import { shallowMount, RouterLinkStub } from '@vue/test-utils';

describe('snapshot songItem.vue', () => {
  test('renders correctly', () => {
    const song = {
      id: 'abc',
      name: 'test',
    };

    const wrapper = shallowMount(SongItem, {
      propsData: {
        song,
      },
      global: {
        components: {
          'router-link': RouterLinkStub,
        },
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
