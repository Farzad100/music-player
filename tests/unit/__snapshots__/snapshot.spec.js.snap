// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot songItem.vue renders correctly 1`] = `
<li
  class="flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-50 snapshot song-id-abc"
>
  <div>
    <router-link-stub
      class="font-bold block text-gray-600"
      custom="false"
      to="[object Object]"
    />
    <span
      class="text-gray-500 text-sm"
    />
  </div>
  <!-- &lt;div class="text-gray-600 text-lg"&gt;
              &lt;router-link custom
              :to="{ name: 'song', params: { id: song.id }, hash: '#comments'}"
              v-slot="{ navigate }"&gt;
                &lt;span class="comments" @click="navigate"&gt;
                  &lt;i class="fa fa-comments text-gray-600"&gt;&lt;/i&gt;
                  {{ comment_count }}
                &lt;/span&gt;
              &lt;/router-link&gt;
            &lt;/div&gt; -->
</li>
`;
